keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -73,255 +76,242 @@ private void verifyEnvironmentVariables() {
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
-
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+            Collections.emptyList(), Collections.emptyList(), AbstractMap.SimpleEntry::new);
+
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    assertThat(secretNames)
+        .contains(
+            secretName1WithTagKey1ValA,
+            secretName2WithTagKey1ValB,
+            secretName3WithTagKey2ValC,
+            secretName4WithTagKey2ValB);
   }
 
   @Test
-  void listAndMapSingleSecretWithMultipleTags() {
-    createSecret(false, true, false);
-
+  void nonExistentTagFiltersReturnsEmpty() {
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
+            List.of("nonexistent-tag-key"),
+            List.of("nonexistent-tag-value"),
             AbstractMap.SimpleEntry::new);
 
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    assertThat(secretNames).isEmpty();
   }
 
   @Test
-  void listAndMapMultipleSecretsWithMultipleTags() {
-    createSecret(true, true, false);
-
+  void listAndMapSecretsWithMatchingTagKeys() {
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
-
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+            List.of("tagKey1"), Collections.emptyList(), AbstractMap.SimpleEntry::new);
+
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    final List<String> secretValues =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getValue()))
+            .collect(Collectors.toList());
+    assertThat(secretNames)
+        .contains(secretName1WithTagKey1ValA, secretName2WithTagKey1ValB)
+        .doesNotContain(secretName3WithTagKey2ValC, secretName4WithTagKey2ValB);
+    assertThat(secretValues).contains(SECRET_VALUE);
```

Comment:
With these all having the same secret value we aren't testing that all the secret values are returned. I think the secret values need to be unique for these to be a valid test.